server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java

```diff
@@ -661,6 +663,97 @@ public void testRemoteClusterOnlyCCSWithFailuresOnAllShards() throws Exception {
         }
     }
 
+    public void testDateMathIndexes() throws ExecutionException, InterruptedException {
+        Map<String, Object> testClusterInfo = setupTwoClusters(
+            new String[] { "datemath-2001-01-01-14", "datemath-2001-01-01-15" },
+            new String[] { "remotemath-2001-01-01-14", "remotemath-2001-01-01-15" }
+        );
+        SearchRequest searchRequest = new SearchRequest(
+            REMOTE_CLUSTER + ":<remotemath-{2001-01-01-13||+1h/h{yyyy-MM-dd-HH|-07:00}}>",
+            "<datemath-{2001-01-01-13||+1h/h{yyyy-MM-dd-HH|-07:00}}>"
+        );
+        searchRequest.setCcsMinimizeRoundtrips(randomBoolean());
+        searchRequest.allowPartialSearchResults(false);
+        searchRequest.source(new SearchSourceBuilder().query(new MatchAllQueryBuilder()).size(5000));
+        assertResponse(client(LOCAL_CLUSTER).search(searchRequest), response -> {
+            assertNotNull(response);
+            Clusters clusters = response.getClusters();
+            assertFalse("search cluster results should NOT be marked as partial", clusters.hasPartialResults());
+            assertThat(clusters.getTotal(), equalTo(2));
+            Cluster localClusterSearchInfo = clusters.getCluster(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
+            assertNotNull(localClusterSearchInfo);
+            Cluster remoteClusterSearchInfo = clusters.getCluster(REMOTE_CLUSTER);
+            assertNotNull(remoteClusterSearchInfo);
+            assertThat(Objects.requireNonNull(response.getHits().getTotalHits()).value, greaterThan(2L));
+            for (var hit : response.getHits()) {
+                assertThat(hit.getIndex(), anyOf(equalTo("datemath-2001-01-01-14"), equalTo("remotemath-2001-01-01-14")));
+            }
+        });
+    }
+
+    /**
+     * Test for issue https://github.com/elastic/elasticsearch/issues/112243
```

Comment:
I am not entirely sure I follow why this is a remote cluster test? IIUC, the original issue will happen in a single cluster setup too? I also think the search only addresses a single cluster. Should we move this to a different IT to make this clear?

Reply:
I need to test that both remote & local scenarios work fine - i.e. that scenarios that include valid remotes with date expressions work, and scenarios that include date expressions that look like remote but aren't work. If I only test it on single cluster, I may have broken the valid remote case but I wouldn't know that, as AFAIK we presently don't have any tests that verify this scenario.

Reply:
Maybe we need tests in two places then? Regular tests to verify the local only scenario, and this one to verify the cross-cluster scenario?